### ISSUE TITLE
[Discord Suggestion] Stash

### ISSUE THREAD

**Issue by Imrielle** [ldtteam/minecolonies#2121]
A new block: Stash
Put in your personal home. Delivery man will come pick up things in it.
Advantages: allows players to interact with the colony's inventory without leaving the house

From: Warpdragon#9105

**Comment by MattBeard** [comment]
Nice, it also means that you can get stuff to the warehouse without needing to figure out where best to put it. Should work like a normal worker chest, but with nothing left behind (it seems that worker chests are allowed to accumulate certain items the worker needs)

It would also be great here, or in another chest-like item, to be able to program in a set of "deliver to me" buttons. These would add requests to the RS like the workers. I envisage that there would be a number of slots (perhaps the number increases with upgrade or each new slot can be bought with something like emeralds). Ideally there would be a number (with up down arrows, or type-able) that says how many you want. You may want one iron pick-axe, but 20 cooked steak, or 64 oak planks.

The request part could be a tab on the chest screen and the DMan would need some way to not remove items you requested (otherwise one DMan could remove the item another just added!) Probably the safest way to achieve this would be for them to arrive in a separate "delivered items" area on that tab rather than adding to the main stash area. Items arriving in the chest would need to check that item on the request page, and decrease the request amount by that number. That means that if you requested 16 planks and the DMan delivered 10, the requested number drops to 6.

**Comment by Raycoms** [comment]
Closed in favor of joint issue

**Comment by Shirkit** [comment]
Ok, as I wanted this feature and it's not currently implemented, I started working on it. It shouldn't take too much for me to finish, as the work is mostly done already. But I have a couple of questions to the devs here:

1. The priority system, I couldn't understand what is it for. Is it for fulfilling requests from the RequestSystem, or to tell the EntityAIWorkDeliveryman to pickup more often? Or is it both?
2. I got stuck while trying to get my `colony.buildings.workerbuilding` class to get notified when the contents of the inventory of the TileEntity is changed. Is there a way to notify my building that the inventory has changed? I saw that the Warehouse manager gets notified, but I know that somehow it notifies, as workers gets to know quite fast when we deposit some item inside his hut inventory.

Thanks
